matchmake-extension/database/get_playing_session.go

```diff
@@ -0,0 +1,93 @@
+package database
+
+import (
+	"database/sql"
+
+	"github.com/PretendoNetwork/nex-go/v2"
+	"github.com/PretendoNetwork/nex-go/v2/types"
+	common_globals "github.com/PretendoNetwork/nex-protocols-common-go/v2/globals"
+	match_making_types "github.com/PretendoNetwork/nex-protocols-go/v2/match-making/types"
+)
+
+// GetPlayingSession returns the playing sessions of the given PIDs
+func GetPlayingSession(manager *common_globals.MatchmakingManager, listPID []types.PID) ([]match_making_types.PlayingSession, *nex.Error) {
+	playingSessions := make([]match_making_types.PlayingSession, 0)
+	for _, pid := range listPID {
+		playingSession := match_making_types.NewPlayingSession()
+		resultMatchmakeSession := match_making_types.NewMatchmakeSession()
+		var resultMatchmakeParam []byte
+
+		err := manager.Database.QueryRow(`SELECT
+			g.id,
+			g.owner_pid,
+			g.host_pid,
+			g.min_participants,
+			g.max_participants,
+			g.participation_policy,
+			g.policy_argument,
+			g.flags,
+			g.state,
+			g.description,
+			array_length(g.participants, 1),
+			g.started_time,
+			ms.game_mode,
+			ms.attribs,
+			ms.open_participation,
+			ms.matchmake_system_type,
+			ms.application_buffer,
+			ms.progress_score,
+			ms.option_zero,
+			ms.matchmake_param,
+			ms.refer_gid,
+			ms.user_password_enabled,
+			ms.system_password_enabled,
+			ms.codeword
+		FROM matchmaking.gatherings AS g
+		INNER JOIN matchmaking.matchmake_sessions AS ms ON ms.id = g.id
+		WHERE
+		g.registered=true AND
+		g.type='MatchmakeSession' AND
+		$1=ANY(g.participants)`, pid).Scan(
```

Comment:
You can get the session info for anyone, but if you request info for a user who has you blocked then that users info is not returned. At the very least we should make a note of this somewhere, if not adding it now